src/test/java/au/org/ala/biocache/controller/DownloadControllerTest.java

```diff
@@ -0,0 +1,138 @@
+package au.org.ala.biocache.controller;
+
+import au.org.ala.biocache.service.AuthService;
+import au.org.ala.biocache.service.DownloadService;
+import au.org.ala.biocache.service.LoggerService;
+import au.org.ala.biocache.web.DownloadController;
+import au.org.ala.biocache.web.OccurrenceController;
+import com.google.common.collect.ImmutableMap;
+import junit.framework.TestCase;
+import org.ala.client.model.LogEventVO;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+/**
+ * Integration tests for occurrence services.
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {"classpath:springTest.xml"})
+@WebAppConfiguration
+public class DownloadControllerTest extends TestCase {
+
+    static {
+        System.setProperty("biocache.config", System.getProperty("user.dir") + "/src/test/resources/biocache-test-config.properties");
+    }
+
+    public final int TEST_INDEX_SIZE = 1000;
+    public final int DEFAULT_SEARCH_PAGE_SIZE = 10;
+    public final int INDEXED_FIELD_SIZE = 377;
+
+    @Autowired
+    DownloadController downloadController;
+
+    @Autowired
+    DownloadService downloadService;
+
+    AuthService authService;
+
+    @Autowired
+    WebApplicationContext wac;
+
+    MockMvc mockMvc;
+
+    @Before
+    public void setup() {
+
+        authService = mock(AuthService.class);
+        ReflectionTestUtils.setField(downloadController, "authService", authService);
+
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
+    }
+
+    @Test
+    public void offlineDownloadValidEmailTest() throws Exception {
+
+        when(authService.getUserDetails("test@test.com"))
+                .thenReturn((Map)ImmutableMap.of(
+                        "locked", false,
```

Comment:
This positive test is failing currently because it doesn't have an "activated=true" in the response. 

From the travis build failure log in `https://travis-ci.org/github/AtlasOfLivingAustralia/biocache-service/builds/721907178`:

```
ERROR] Failures: 

[ERROR]   DownloadControllerTest.offlineDownloadValidEmailTest:86 Status expected:<200> but was:<403>
```

It would also be useful to have an "activated=false,locked=false,roles=[ROLE_USER]" negative test.

It would also be useful to add "activated=true" to the tests below to make sure they aren't just failing because "activated" isn't there.

Reply:
The error for this changed from `403` to `500`, but the test is still failing:

```
[ERROR] Failures: 

[ERROR]   DownloadControllerTest.offlineDownloadValidEmailTest:87 Status expected:<200> but was:<500>
```

Reply:
The error `500` is due to a filesystem permission issue with running tests in Travis. The download integration test need to write the status file to `/data/cache/downloads`. 
I'll look into mocking the filesystem access.

Reply:
Have made fixed to the `DownloadControllerTest` to change location of directories under `/data` to `org.junit.rules.TemporaryFolder`. All tests are now working again.